### PR TITLE
[actions] Run container and linter tests on PRs

### PR DESCRIPTION
**Pull request by black-dragon74**
/cc: @nixpanic @mscherer 

Signed-off-by: black-dragon74 <nickk.2974@gmail.com>

### CODE REVIEW

File: tests/container-tests.sh [hunk 1]
```diff
@@ -4,6 +4,17 @@
...
 
+# Allow script to be executed both by abs and relative paths
+cd $(dirname $0)
+
+# GH Actions doesn't support TTY but one may manually run this script locally
+# Add conditional TTY support
+test -t 1 && \
+    {
+        USE_TTY="-t"
+        echo "TTY support is 1"
```
[the remark this answers is not in the document]
Comment (black-dragon74) [reply tests/container-tests.sh:15]: Ah, that's an artifact of testing. Will remove it. Should be
```sh
test -t 1 && USE_TTY="t"
```

File: .github/workflows/lint-and-test-pr.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      # Checkout the current repo
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      # Prepare the runner
+      - name: Prepare the runner
+        run: |
+          sudo apt install shellcheck
+          sudo npm install -g dockerfile_lint
+          chmod +x tests/container-tests.sh
```
Comment (nixpanic) [inline .github/workflows/lint-and-test-pr.yml:21]: you can drop this `chmod` if the file in the repository is executable, which I think it is?

**Review by nixpanic — APPROVED** [review]
automated tests on PRs, that's great!

File: .github/workflows/lint-and-test-pr.yml (hunk 2 above)
Comment (humblec) [reply .github/workflows/lint-and-test-pr.yml:21]: @black-dragon74 please address above. Rest looks good to me

**Pull request merged**